triplet/main.py

```diff
@@ -230,6 +238,31 @@ def train(epoch):
                 100. * batch_idx / len(train_loader), loss.data[0]))
 
 
+def test(epoch):
+    model.eval()
+    scores, distances = [], []
+    for batch_idx, (data_a, data_p, label) in enumerate(test_loader):
+        if args.cuda:
+            data_a, data_p = data_a.cuda(), data_p.cuda()
+        data_a, data_p, labels = Variable(data_a, volatile=True), \
+                                 Variable(data_p, volatile=True), Variable(label)
+        out_a, out_p = model(data_a), model(data_p)
+        dists = torch.sqrt(torch.sum((out_a - out_p) ** 2, 1))  # euclidean distance
+        distances.extend(dists)
```

Comment:
@cumttang Good catch. The problem in here, is that models return `Variable`, while you want to work with `Tensors` (that can be obtained with `.data`). Also, are you sure you want to use `.extend` and not `.append`? It will iterate over the tensors, which can be a bit slow.